File: solutions/java/isbn-verifier/1/src/main/java/IsbnVerifier.java
import java.lang.String;

class IsbnVerifier {

    boolean isValid(String stringToVerify) {
        String num = stringToVerify.replaceAll("-","");
        int suma = 0;

        if (num.length() == 10) {
            for ( int i = 0; i < 10; i++) {
                if (num.charAt(i) == 'X' && i == 9) {
                    suma = suma + (10-i) * 10;
                } else if (Character.isDigit(num.charAt(i))) {
                    suma = suma + (10 -i)* Character.getNumericValue(num.charAt(i));
                }
                else {
                    return false;
                    }
                    }
            return (suma%11 == 0);
        } else {
            return false;
            }
    }

}

File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    numbers = "".join(isbn.split("-"))
    if len(numbers) != 10:
        return False
    
    suma = 0
    for i in range(10, 1, -1):
        if numbers[10-i].isdigit():
            suma += i * int(numbers[10-i])
        else:
            return False
    if numbers[9] == 'X':
        suma += 10
    elif numbers[9].isdigit():
        suma += int(numbers[9])
    else:
        return False
    return suma % 11 == 0